google_analytics/notebooks/load.ipynb

```diff
@@ -34,24 +34,189 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 64,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loaded file ../data/train.csv\n",
+      "Shape is: (903653, 55)\n"
+     ]
+    }
+   ],
+   "source": [
+    "train = load(\"../data/train.csv\")\n",
+    "\n",
+    "const_cols = [c for c in train.columns if train[c].nunique(dropna=False) == 1]\n",
+    "train = train.drop(const_cols, axis=1)\n",
+    "\n",
+    "# Cast target\n",
+    "train[\"transactionRevenue\"] = train[\"transactionRevenue\"].fillna(0).astype(float)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hypothesis: certain profiles should be naively assigned a zero predicted revenue\n",
+    "\n",
+    "Since the new problem description includes much less information at prediction time (no session info) it might be beneficial to only focus on visitors with a high probability to spend - and assign a zero prediction to everyone else.\n",
+    "\n",
+    "**Which profile is improbable enough to make a purchase, that it makes sense to make that prediction naively?**\n",
+    "\n",
+    "We think that visitors with a single visit that did not lead to a purchase during the training period (prior to December of the testing year) have a very high probability of not buying anything during December and January either. Let's test this assumption."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 65,
    "metadata": {},
    "outputs": [],
    "source": [
-    "train = load(\"../data/train.csv\", nrows=100000)\n",
-    "test = load(\"../data/test.csv\", nrows=10000)\n",
+    "x_train = train[(train[\"date\"] < '2016-12-01')]\n",
```

Comment:
Our X_test data is from August 2017 until Oktober 15 2018.  Shouldn't our X_train be the same time period, but then the year before to make it a fair comparison? So, in this case, the period of X_train is August 2016 until Oktober 15, 2017 (date < 2017-10-15).